cdgriffith_pyfiledrop_pyfiledrop_33.py

```
@@ -0,0 +1,277 @@
+    if not dz_uuid:
+        # Assume this file has not been chunked
+        with open(storage_path / f"{uuid.uuid4()}_{secure_filename(file.filename)}", "wb") as f:
+            file.save(f)
+        return "File Saved"
+
+    # Chunked download
+    try:
+        current_chunk = int(request.forms["dzchunkindex"])
+        total_chunks = int(request.forms["dztotalchunkcount"])
+    except KeyError as err:
+        raise HTTPError(status=400, body=f"Not all required fields supplied, missing {err}")
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)